code-challenges.js

```diff
@@ -32,8 +67,102 @@ var hand4 = [3, 5, 3, 5, 3]
+//     let value2 =
+//     value1 != value2
+//     //map over the array to check if the value also exists at every other value in the array.
+//     //If it does, add one to the 3 of a kind variable for each time it occurs in the array.
+//     array.map(value => {
+//         if (possible3OfAKind.indexOf(value) === -1){
+//             return possible3OfAKind++
+//         } else if (possiblePair.indexOf(value) === -1) {
+//             return possiblePair++
+//         }
+//         //the main issue that i see here is that there is not a way to check if the values are the same. in cards there is no way for all 5 values to be the same because there are only 4 of each card. 
+//     })
+//     if (possible3OfAKind === 3 && possiblePair === 2){
+//     return "Full House!"
+//     } else {
+//         return "Not a full house try again"
+//     }
+// }
+
+// i ran into the issue where I wanted to create a function that could check this for ANY value but it was hard to do unless you know the values you want to check for. 
+
+
+
 // STRETCH:
 var hand5 = ["K", 5, "K", 5, "K"]
 // Expected output: true
 var hand6 = ["A", "Q", "A", 5, 3]
 // Expected output: false
+
+// //arrays of cards? i feel like i might need these?
+// let card1 = ["A", 2, 3, 4, 5, 6, 7, 8,  9, 10, "J", Q, K]
+// let card2 = ["A", 2, 3, 4, 5, 6, 7, 8,  9, 10, "J", Q, K]
+
+
+const fullHouse = (array) => {
+    //set variables for some empty arrays where we will push the values if they are duplicates.
+    let firstCardArray = []
+    let secondCardArray = []
+    let thirdCardArray = []
+    let fourthCardArray = []
+    //map over the array and take the value as an argument
+    array.map(value => {
+        //if the value is the first value in the array or is equal to the first value in the array,
+        if(value === array[0]){
+            //push it to firstCardArray so we can keep count of how many times it occurs in the hand
+            return firstCardArray.push(value)
+        //if the value does not equal the first value in the array, if it equals the second value in the array or if it is the same as the second value in the array
+        }else if((value === array[1]) && (value != array[0])){
+            //push it to the secondCardArray so we can then count the number of times it occurs
+            return secondCardArray.push(value)
+        //if the value does not equal the first value in the array, if it is a new value and equals the third value in the array or if it is the same as the third value in the array
+        }else if((value === array[2]) && (value != array[0])) {
+            return thirdCardArray.push(value)
+        //do the same thing for the 4th value
+        }else if((value === array[3]) && (value != array[0])){
+            return fourthCardArray.push(value)
+        }
+    })
+    //if the lengths of 2 of the arrays are 2 & 3 then the hand will be a full house.
+    if((firstCardArray.length === 2 && secondCardArray.length === 3) ||
```

Comment:
General thing to keep in mind is that anytime an if statement is made with many (~more than 2) conditionals, there is likely a different way to model data so that don't require as many conditionals.